### DISCUSSION TITLE
Cloning Amazon.Route53.Model.ResourceRecordSet and/or Amazon.Route53.Model.ResourceRecord objects

### DISCUSSION THREAD

**Discussion by mcdonamw** [aws/aws-sdk-net#3350] [Q&A] [answered]
I'm working on a script to update some Route53 resource records and am having a difficult time.  I'm currently pulling a recordset, modifying the resourcerecords within and resubmitting back to Route53 and this is working.  But I'm trying to instead create a separate object to modify in order to keep the original records for later comparison/reporting, but I'm having a very difficult time copying/cloning these object types.

It is my understanding neither of these object types implement the System.ICloneable interface.  Is that by design?

At the moment I'm planning to do the following but not sure if this will work or how scalable it will be for other resource record types (I'm working with TXT records atm).

```
$NewR53ResourceRecordSet = New-Object Amazon.Route53.Model.ResourceRecordSet
$NewR53ResourceRecordSet.Name = $CurrentR53ResourceRecordSet.Name
$NewR53ResourceRecordSet.Type = $CurrentR53ResourceRecordSet.Type
$NewR53ResourceRecordSet.TTL = $CurrentR53ResourceRecordSet.TTL
foreach ($CurrentR53ResourceRecord in $CurrentR53ResourceRecordSet.ResourceRecords) {
    $NewR53ResourceRecord = New-Object Amazon.Route53.Model.ResourceRecord
    $NewR53ResourceRecord.Value = $CurrentR53ResourceRecord.Value
    $NewR53ResourceRecordSet.ResourceRecords += $NewR53ResourceRecord
}
```

Is there a better way to do something like this?  Also is it documented somewhere which properties are required to submit in a change batch to be successful?  Here I'm taking the Name, Type, TTL, and ResourceRecords but I'm only changing the Resource Records so I'm wondering if I need to capture the others.

**Comment by bhoradc** [comment, marked as answer]
Hi @mcdonamw, 

Thank you for opening this discussion. This looks like a potential feature request. 

Would you be interested in submitting a `Feature Request` under the [AWS Tools for PowerShell](https://github.com/aws/aws-tools-for-powershell) repo? 

We would love to hear from more users who share your need. This will help us gauge interest and prioritize future development efforts. 

Regards, 
Chaitanya
